src/Page.js

```diff
@@ -530,6 +532,33 @@ Page.prototype.removeFrontMatter = function (includedPage) {
   return $.html();
 };
 
+/**
+  * Inserts the page layout's header to the start of the page
+  * @param pageData a page with its front matter collected
+  */
+Page.prototype.insertHeaderFile = function (pageData) {
+  const { header } = this.frontMatter;
+  let headerFile;
+  if (header) {
+    headerFile = path.join(HEADERS_FOLDER_PATH, header);
+  } else {
+    headerFile = path.join(LAYOUT_FOLDER_PATH, this.frontMatter.layout, LAYOUT_HEADER);
+  }
+  const headerPath = path.join(this.rootPath, headerFile);
+  if (!fs.existsSync(headerPath)) {
+    return pageData;
```

Comment:
Since headers and footers are optional, an explicit error should not be logged. 

To add on, other optional items such as `site-nav` and `page-nav` are also not logged if they are not added. Furthermore, if we do log it then the console will be filled with messages if there are a significant amount of pages (with missing header files).

Although I can see explicit logging useful for debugging purposes. E.G.

```console
header file skipped: header file not found
page navigation added: {Title: Test Title} {headingIndexingLevel: 2}
footer file skipped: footer not specified, layout not specified
```
But alas, not in the scope of this PR :)